lib/tests/streamlit/elements/file_uploader_test.py

```diff
@@ -101,25 +112,20 @@ def test_multiple_files(self, get_file_recs_patch):
             # If "accept_multiple_files" is True, then we should get a list of
             # values back. Otherwise, we should just get a single value.
 
-            # Because file_uploader returns unique UploadedFile instances
-            # each time it's called, we convert the return value back
-            # from UploadedFile -> UploadedFileRec (which implements
-            # equals()) to test equality.
-
             if accept_multiple:
-                results = [
-                    UploadedFileRec(file.id, file.name, file.type, file.getvalue())
-                    for file in return_val
-                ]
-                self.assertEqual(file_recs, results)
+                self.assertEqual(return_val, uploaded_files)
+
+                for actual, expected in zip(return_val, uploaded_files):
+                    self.assertEqual(actual.name, expected.name)
+                    self.assertEqual(actual.type, expected.type)
+                    self.assertEqual(actual.size, expected.size)
+                    self.assertEqual(actual.getvalue(), expected.getvalue())
             else:
-                results = UploadedFileRec(
-                    return_val.id,
-                    return_val.name,
-                    return_val.type,
-                    return_val.getvalue(),
-                )
-                self.assertEqual(file_recs[0], results)
+                self.assertEqual(return_val, uploaded_files[0])
```

Comment:
nit: probably worth pulling `uploaded_files[0]` out into an intermediary variable to avoid having to index into the list repeatedly